src/random/XorwowRngEngine.hh

```diff
@@ -0,0 +1,114 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file XorwowRngEngine.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Assert.hh"
+#include "base/OpaqueId.hh"
+#include "base/Types.hh"
+#include "random/distributions/GenerateCanonical.hh"
+
+#include "XorwowRngData.hh"
+#include "detail/GenerateCanonical32.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Generate random data using the XORWOW algorithm.
+ *
+ * The XorwowRngEngine uses a C++11-like interface to generate random data. The
+ * sampling of uniform floating point data is done with specializations to the
+ * GenerateCanonical class.
+ *
+ * This class does not define an initializer because it is assumed that the
+ * state has been fully randomized at initialization (see the \c resize
+ * function for \c XorwowRngStateData.)
+ *
+ * See Marsaglia (2003) for the theory underlying the algorithm and the the
+ * "example" \c xorwow that combines an \em xorshift output with a Weyl
+ * sequence.
+ *
+ * https://www.jstatsoft.org/index.php/jss/article/view/v008i14/916
+ */
+class XorwowRngEngine
+{
+  public:
+    //!@{
+    //! Type aliases
+    using result_type = unsigned int;
+    using StateRef = XorwowRngStateData<Ownership::reference, MemSpace::native>;
+    //!@}
+
+  public:
+    // Construct from state
+    inline CELER_FUNCTION
+    XorwowRngEngine(const StateRef& state, const ThreadId& id);
+
+    // Sample a random number
+    inline CELER_FUNCTION result_type operator()();
+
+  private:
+    XorwowState* state_;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Specialization of GenerateCanonical for XorwowRngEngine.
+ */
+template<class RealType>
+class GenerateCanonical<XorwowRngEngine, RealType>
+{
+  public:
+    //!@{
+    //! Type aliases
+    using real_type   = float;
```

Comment:
Sneaky 😉
```suggestion
    using real_type   = RealType;
```

Reply:
🤦‍♂️thanks... Amanda to the rescue again!